### PR TITLE
Exclude certain test suites per spec from running in the Jenkins pipeline

### LINKED ISSUE #5459: Jenkins: Feature Request: Add ability to include/exclude test suites per spec
Add an option to include and or exclude running entire test suites per spec in the defaults.yml variable file.

Ex usage: The CMake builds do not currently support testing so we've had to temporarily create empty "hello world" test jobs so the pipeline call doesn't fail.
Ref #5447

### PR DESCRIPTION
**Pull request by samuel-rubin**
Certain specs do not have any tests associated with them.
This would not allow these tests to run if they are selected
This will not allow any testing on x86-64_linux_cm

[skip ci]
Resolves #5459
Signed-off-by: Samuel Rubin <samuel.rubin@ibm.com>

### CODE REVIEW

**Comment by samuel-rubin** [comment]
My testing: 
sanity.functional for cmake: https://ci.eclipse.org/openj9/job/Sam-Pipeline_Build_Test_JDK11_x86-64_linux_cm/1/console
sanity.functional for ppcle (comparison): https://ci.eclipse.org/openj9/job/Sam-Pipeline_Build_Test_JDK11_ppc64le_linux/1/console

**Comment by samuel-rubin** [comment]
Can @vsebe or @AdamBrousseau please review this? Thank you

File: buildenv/jenkins/common/pipeline-functions [hunk 1]
```diff
@@ -364,6 +364,11 @@ def workflow(SDK_VERSION, SPEC, SHAS, OPENJDK_REPO, OPENJDK_BRANCH, OPENJ9_REPO,
         echo "Using VENDOR_TEST_REPOS = ${VENDOR_TEST_REPOS}, VENDOR_TEST_BRANCHES = ${VENDOR_TEST_BRANCHES}, VENDOR_TEST_SHAS = ${VENDOR_TEST_SHAS}, VENDOR_TEST_DIRS = ${VENDOR_TEST_DIRS}, BUILD_LIST = ${BUILD_LIST}" 
 
         for (name in TARGET_NAMES) {
+            // Checking to see if the test should be excluded
+            if (EXCLUDED_TESTS.contains(name)){
```
Comment (vsebe) [inline buildenv/jenkins/common/pipeline-functions]: EXCLUDED_TESTS is comma separated strings not an array.

Comment (vsebe) [inline buildenv/jenkins/common/pipeline-functions]: EXCLUDED_TESTS is a string not an array.

**Comment by samuel-rubin** [comment]
Testing to see if all the test targets will be skipped: https://ci.eclipse.org/openj9/job/Sam-Pipeline_Build_Test_JDK11_x86-64_linux_cm/4/
Testing if sanity.functional is skipped: https://ci.eclipse.org/openj9/job/Sam-Pipeline_Build_Test_JDK11_x86-64_linux_cm/3/

File: buildenv/jenkins/common/pipeline-functions (hunk 1 above)
Comment (samuel-rubin) [reply buildenv/jenkins/common/pipeline-functions]: I added a tokenize to EXCLUDED_TESTS in the variable-functions, so it is now be an array

**Comment by samuel-rubin** [comment]
I misspelled the test targets, so the previous job failed. Here is the retry: 
https://ci.eclipse.org/openj9/job/Sam-Pipeline_Build_Test_JDK11_x86-64_linux_cm/9/console

File: buildenv/jenkins/common/pipeline-functions (hunk 1 above)
Comment (samuel-rubin) [reply buildenv/jenkins/common/pipeline-functions]: I changed the excluded_tests in the variable file to be a list instead of tokenizing the string.

**Comment by samuel-rubin** [comment]
Here is my test with the excluded tests as a string in the variables file: https://ci.eclipse.org/openj9/job/Sam-Pipeline_Build_Test_JDK11_x86-64_linux_cm/10/console

**Comment by vsebe** [comment]
@pshipton ready for you review, thx

**Pull request merged**